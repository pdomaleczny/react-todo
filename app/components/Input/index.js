import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TextField from 'material-ui/TextField';

class Input extends Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
    };
  }

  keyPressed(event) {
    if (event.key === 'Enter') {
      this.props.addNewTask(this.state.text);
      this.setState({ text: '' });
    }
  }

  render() {
    return (
      <div>
        <TextField
          floatingLabelText='New task'
          value={this.state.text}
          onChange={(event, value) => { this.setState({ text: value }); }}
          onKeyPress={(event) => { this.keyPressed(event); }}
        />
      </div>
    );
  }
}

Input.propTypes = {
  addNewTask: PropTypes.func.isRequired,
};

export default Input;
