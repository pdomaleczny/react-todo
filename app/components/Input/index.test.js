/* eslint-disable */
import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { shallow } from 'enzyme';
import Input from '../Input';

it('should render a input', () => {
    const wrapper = shallow(
        <Input addNewTask={(newTask) => { return newTask; }}/>
    );
    expect(wrapper).toMatchSnapshot();
});


it('should trigger adding task action on enter', () => {
    const addNewTaskMock = jest.fn();
    const wrapper = shallow(
        <Input addNewTask={(newTask) => { addNewTaskMock(); }}/>
    );

    wrapper.instance().keyPressed({ key: 'Enter' })

    expect(wrapper.state('text')).toEqual('');
    expect(addNewTaskMock.mock.calls.length).toEqual(1);
});

it('should trigger adding task action on button click', () => {
    const addNewTaskMock = jest.fn();
    const wrapper = shallow(
        <Input addNewTask={(newTask) => { addNewTaskMock(); }}/>
    );

    wrapper.instance().setState({ text: 'test' });
    wrapper.find('FlatButton').simulate('touchTap');

    expect(wrapper.state('text')).toEqual('');
    expect(addNewTaskMock.mock.calls.length).toEqual(1);
});
