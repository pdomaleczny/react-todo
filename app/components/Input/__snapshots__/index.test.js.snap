// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a input 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="input-container"
>
    <TextField
        disabled={false}
        floatingLabelFixed={false}
        floatingLabelText="New task"
        fullWidth={false}
        multiLine={false}
        onChange={[Function]}
        onKeyPress={[Function]}
        rows={1}
        type="text"
        underlineShow={true}
        value=""
    />
</div>,
  "nodes": Array [
    <div
      className="input-container"
>
      <TextField
            disabled={false}
            floatingLabelFixed={false}
            floatingLabelText="New task"
            fullWidth={false}
            multiLine={false}
            onChange={[Function]}
            onKeyPress={[Function]}
            rows={1}
            type="text"
            underlineShow={true}
            value=""
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Input
        addNewTask={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Input {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "addNewTask": [Function],
        },
        "refs": Object {},
        "state": Object {
          "text": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="input-container"
>
          <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    floatingLabelText="New task"
                    fullWidth={false}
                    multiLine={false}
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    rows={1}
                    type="text"
                    underlineShow={true}
                    value=""
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="input-container"
>
          <TextField
                    disabled={false}
                    floatingLabelFixed={false}
                    floatingLabelText="New task"
                    fullWidth={false}
                    multiLine={false}
                    onChange={[Function]}
                    onKeyPress={[Function]}
                    rows={1}
                    type="text"
                    underlineShow={true}
                    value=""
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Input
    addNewTask={[Function]}
/>,
}
`;
